source/dub/packagesuppliers/filesystem.d

```diff
@@ -61,7 +62,10 @@ class FileSystemPackageSupplier : PackageSupplier {
 		string packageFileContent = packageInfoFileFromZip(filePath, packageFileName);
 		auto recipe = parsePackageRecipe(packageFileContent, packageFileName);
 		Json json = toJson(recipe);
-		json["version"] = filePath.toNativeString().split("-")[$-1].stripExtension();
+		auto basename = filePath.head.name;
+		enforce(basename[$-4..$] == ".zip", "Malformed package filename: " ~ filePath.toNativeString);
```

Comment:
use endsWith for consistency pls (and to avoid potential range errors, though they cannot really occur now)